cmd/maya-apiserver/cstor-operator/controller.go

```diff
@@ -71,112 +67,115 @@ type Controller struct {
-		// 1. In case the object is updated ( Change of Resource Version)
-
-		// 2. In case the object is deleted by using a finalizer.
-		//    Some of the kubectl version e.g. v1.11.0 will make use of a finalizer beforehand deletion.
-		//    Hence in this case the delete of an object will cause a invoke of UpdateFunc
-		//    as the finalizer addition will cause Resource Version to change.
-		//
-		//    But some of the kubectl version will not make use of finalizer for
-		//    object deletion.
-		//    Whatever be the case, delete of the resource will always invoke DeleteFunc
-		//    and hence the delete event will be captured by it and the delete event
-		//    that will be delivered in UpdateFunc due to specific kubectl versions will
-		//    be suppressed.(see in updateSpc function, where delete event is marked as ignoreEvent)
-
-		// 3. After every fixed amount of time which is know as reSync Period.
-		//    ReSync period can be set to values we want. It can help in reconiciliation.
-		UpdateFunc: controller.updateSpc,
+		AddFunc:    cb.Controller.addSpc,
+		UpdateFunc: cb.Controller.updateSpc,
+		// This will enter the sync loop and no-op, because the spc has been deleted from the store.
+		DeleteFunc: cb.Controller.deleteSpc,
 	})
+	return cb
+}
 
-	return controller
+// Build returns a controller instance.
+func (cb *ControllerBuilder) Build() *Controller {
+	return cb.Controller
 }
 
+// addSpc is the add event handler for spc.
 func (c *Controller) addSpc(obj interface{}) {
 	spcObject := obj.(*apis.StoragePoolClaim)
-	c.queueLoad.Operation = addEvent
-	c.queueLoad.Object = spcObject
 	glog.V(4).Infof("Queuing SPC %s for add event", spcObject.Name)
-	c.enqueueSpc(&c.queueLoad)
+	c.enqueueSpc(obj)
 }
 
+// updateSpc is the update event handler for spc.
 func (c *Controller) updateSpc(oldSpc, newSpc interface{}) {
 	spcObjectNew := newSpc.(*apis.StoragePoolClaim)
 	spcObjectOld := oldSpc.(*apis.StoragePoolClaim)
-
-	if spcObjectNew.ObjectMeta.ResourceVersion == spcObjectOld.ObjectMeta.ResourceVersion {
-		// If Resource Version is same it means the object has not got updated.
-		// Enqueue the object as part of sync event to achieve the reconciliation loop for storagepool
-		// Reconciliation will try to converge the pool to its desired state
-		// If the storagepool is already in the desired state, it will do nothing.
-		c.queueLoad.Operation = syncEvent
-		c.queueLoad.Object = spcObjectNew
-		c.enqueueSpc(&c.queueLoad)
-	} else {
-		if IsDeleteEvent(spcObjectNew) {
-			// Ignore the delete event
-			// We can enqueue object in future if we need to do some other task in case of delete event.
-			c.queueLoad.Operation = ignoreEvent
-		} else {
-			// To-DO
-			// Implement Logic for Update of SPC object
-			c.queueLoad.Operation = updateEvent
-			c.queueLoad.Object = spcObjectNew
-			glog.V(4).Infof("Queuing SPC %s for update event", spcObjectNew.Name)
-			c.enqueueSpc(&c.queueLoad)
-		}
-
+	// TODO: Use CSP controller instance
```

Comment:
please add comments with the logic we want to do the enqueue. Seems few more cases need the enqueue of newSPC